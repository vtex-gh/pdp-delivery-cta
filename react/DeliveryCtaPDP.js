import DeliveryCtaPDP from './components/DeliveryCtaPDP/index'

export default DeliveryCtaPDP
