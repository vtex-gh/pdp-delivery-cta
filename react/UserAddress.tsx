import React, { useEffect } from 'react'
import { useQuery } from 'react-apollo'
import { useCssHandles } from 'vtex.css-handles'
import type { CssHandlesTypes } from 'vtex.css-handles'
import ADDRESS_QUERY from 'vtex.store-resources/QueryAddress'

import Container from './Container'
import AddressInfo, {
  CSS_HANDLES as AddressInfoCssHandles,
} from './components/UserAddress/AddressInfo'
import { UserAddressCssHandlesProvider } from './components/UserAddress/UserAddressCssHandles'

export const USER_ADDRESS_CSS_HANDLES = [
  'userAddressContainer',
  ...AddressInfoCssHandles,
] as const

type Props = {
  /** Define if address should be displayed in a single line or in blocks */
  variation: 'inline' | 'bar'
  /** Define if street should be displayed */
  showStreet?: boolean
  /** Define if city and state should be displayed */
  showCityAndState?: boolean
  /** Define if postal code should be displayed */
  showPostalCode?: boolean
  /** Define if prefix should be displayed */
  showPrefix?: boolean
  /** Define if component should render if address is empty */
  showIfEmpty?: boolean
  /** Used to override default CSS handles */
  classes?: CssHandlesTypes.CustomClasses<typeof USER_ADDRESS_CSS_HANDLES>
}

function UserAddress({
  variation,
  showStreet = true,
  showCityAndState = false,
  showPostalCode = false,
  showPrefix = true,
  showIfEmpty = false,
  classes,
}: Props) {
  const { handles, withModifiers } = useCssHandles(USER_ADDRESS_CSS_HANDLES, {
    classes,
  })

  // todo: type the data returned
  const { data: orderFormData, refetch } = useQuery(ADDRESS_QUERY, {
    ssr: false,
  })

  useEffect(() => {
    const handleLocationUpdated = () => refetch()

    window.addEventListener('locationUpdated', handleLocationUpdated)

    return () => {
      window.removeEventListener('locationUpdated', handleLocationUpdated)
    }
  }, [refetch])

  const orderForm = orderFormData?.orderForm

  if (!orderForm || (!orderForm?.shippingData?.address && !showIfEmpty)) {
    return null
  }

  const isInline = variation === 'inline'
  const isInverted = !isInline

  const AddressInfoWithHandles = () => {
    return (
      <UserAddressCssHandlesProvider
        handles={handles}
        withModifiers={withModifiers}
      >
        <AddressInfo
          inverted={isInverted}
          inline={isInline}
          orderForm={orderForm}
          showStreet={showStreet}
          showCityAndState={showCityAndState}
          showPostalCode={showPostalCode}
          showPrefix={showPrefix}
          showIfEmpty={showIfEmpty}
        />
      </UserAddressCssHandlesProvider>
    )
  }

  if (isInline) {
    return (
      <div
        className={`${handles.userAddressContainer} ph5`}
        style={{
          maxWidth: '30rem',
        }}
      >
        <AddressInfoWithHandles />
      </div>
    )
  }

  return (
    <div
      className={`${handles.userAddressContainer} bg-base--inverted c-on-base--inverted flex ph5 pointer pv3 ml3 mr3`}
    >
      <Container className="flex justify-center w-100 left-0">
        <div className="w-100 mw9 flex">
          <AddressInfoWithHandles />
        </div>
      </Container>
    </div>
  )
}

export default UserAddress
