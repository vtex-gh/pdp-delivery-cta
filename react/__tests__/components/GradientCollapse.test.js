import React from 'react'
import { render } from '@vtex/test-tools/react'

import GradientCollapse from '../../GradientCollapse'

describe('<GradientCollapse /> component', () => {
  function renderComponent(customProps = {}) {
    const props = {
      collapseHeight: 50,
      ...customProps,
    }

    const component = <GradientCollapse {...props}>Test</GradientCollapse>

    return render(component)
  }

  it('should be able to mount and not break', () => {
    const { asFragment } = renderComponent()

    expect(asFragment()).toBeTruthy()
  })

  it('should match snapshot', () => {
    const { asFragment } = renderComponent()

    expect(asFragment()).toMatchSnapshot()
  })
})
