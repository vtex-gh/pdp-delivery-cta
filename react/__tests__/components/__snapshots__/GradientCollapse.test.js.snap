// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GradientCollapse /> component should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="container relative"
    style="transition: 600ms ease-in-out; height: auto; overflow: hidden; display: block;"
  >
    <div
      class="content h-auto"
    >
      Test
    </div>
    <div
      class="pointerEventsNone dn absolute bottom-0 w-100 h-100 flex-column justify-end"
    >
      <div
        class="fadeBottom w-100 h-50"
        style="transition: 400ms ease-in-out;"
      />
      <div
        class="pointerEventsAuto bg-base tc w-100"
      >
        <button
          class="showMoreButton c-action-primary t-action pointer ma5 bn outline-0"
        >
          Show more
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
